### ISSUE TITLE
Generating reports using Reporter fails (IndexError in rt_errors_parse_xml)

### ISSUE THREAD

**Issue by ArekKuczynski** [damies13/rfswarm#304]
**Describe the bug**
This bug was found during regression testing Reporter. Looks like this bug refers to only Linux and macOS.
Action: https://github.com/NiceProjectPoland/rfswarm/actions/runs/11114501282
The test run only DOCX report file generation but it's probably occurring also for HTML and XLSX.
```
Exception in thread Thread-5 (cg_export_word):
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 9923, in cg_export_word
    core.export_word()
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 3307, in export_word
    self.docx_add_sections("TOP", sectionpct)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 4366, in docx_add_sections
    self.docx_add_sections(sect, sectionpct)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 4359, in docx_add_sections
    self.docx_sections_errors(id)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 4795, in docx_sections_errors
    base.rt_errors_get_data(id)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 2399, in rt_errors_get_data
    base.rt_errors_parse_xml(id, rid)
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/rfswarm_reporter/rfswarm_reporter.py", line 2451, in rt_errors_parse_xml
    failkw = failkws[-1]
             ~~~~~~~^^^^
IndexError: list index out of range
```

**To Reproduce**
1. Open Reporter
2. Load Results and Template (used template and results are in Tests/Regression/Reporter/testdata/Issue-#36_37_38
3. Generate DOCX

**Expected behavior**
Report files should be generate without any exceptions.

**Screenshots**

**Desktop (please complete the following information):**
- OS: Linux, macOs
- Reporter Version: v1.4.0 newest 30/9/2024

**Additional context**

**Comment by ArekKuczynski** [comment]
This Issue could be related to the Issue #303 because this problem is no longer seen after implementing bugfix for that Issue

**Comment by damies13** [comment]
I expect this was cause by the bug in the file glob patterns that was selecting the wrong xml file and therefore not finding any matches in the xml it opened. 

This would explain why your not seeing this anymore and would make it hard to reproduce or create tests for now.

I think as a safety measure I should first add an if statement to only get the last item from failkws (`failkws[-1]`) if there is data in failkws, and also prevent the following steps from failing, so I'll submit a fix for this even though it's unlikely to happen again. But I don't think we can make a test case for this.

**Comment by damies13** [comment]
Fix merged into the v1.4.0 branch
